uniffi_bindgen/src/bindings/swift/templates/Helpers.swift

```diff
@@ -0,0 +1,108 @@
+
+
+{% if ci.iter_callback_interface_definitions().len() > 0 %}
+
+class Lock {
+    init(){}
+    func withLock<T>(f: () throws -> T ) rethrows -> T {
+        objc_sync_enter(self)
+        defer { objc_sync_exit(self) }
+        return try f()
+    }
+}
+
+typealias Handle = UInt64
+class ConcurrentHandleMap<T: AnyObject> {
+    private var leftMap: [Handle: T] = [:]
+    private var counter: [Handle: UInt64] = [:]
```

Comment:
Reference counting is done here for callbacks. In rust it is done in FFIObject and for all objects.